### ISSUE TITLE
no spaces in login message

### ISSUE THREAD

**Issue by mikhailnov** [linux-pam/linux-pam#151]
```
msgid "Last login:%s%s%s"
msgstr "Последний вход в систему:%s%s%s"
```
https://github.com/linux-pam/linux-pam/blob/master/po/ru.po#L308

It results to text without spaces:
```
Последняя неудачная попытка входа в систему:Вс ноя  3 21:20:57 MSK 2019на console
```

There must be spaces: `Last login: %s %s %s`

**Comment by ldv-alt** [comment]
On Sun, Nov 03, 2019 at 11:10:05AM -0800, mikhailnov wrote:
> ```
> msgid "Last login:%s%s%s"
> msgstr "Последний вход в систему:%s%s%s"
> ```
> https://github.com/linux-pam/linux-pam/blob/master/po/ru.po#L308
> 
> It results to text without spaces:
> ```
> Последняя неудачная попытка входа в систему:Вс ноя  3 21:20:57 MSK 2019на console
> ```
> 
> There must be spaces: `Last login: %s %s %s`

No, the msgid you are talking about is correct (and the msgstr is correct,
too), you can tell this by inspecting the source code in
modules/pam_lastlog/pam_lastlog.c file.

The actual bug is in another part of po/ru.po that incorrectly translates
the following format strings:

#. TRANSLATORS: "strftime options for date of last login"
#: modules/pam_lastlog/pam_lastlog.c:282 modules/pam_lastlog/pam_lastlog.c:496
msgid " %a %b %e %H:%M:%S %Z %Y"
msgstr "%a %b %e %H:%M:%S %Z %Y"

#. TRANSLATORS: " from <host>"
#: modules/pam_lastlog/pam_lastlog.c:291 modules/pam_lastlog/pam_lastlog.c:505
#, c-format
msgid " from %.*s"
msgstr "с %.*s"

#. TRANSLATORS: " on <terminal>"
#: modules/pam_lastlog/pam_lastlog.c:303 modules/pam_lastlog/pam_lastlog.c:517
#, c-format
msgid " on %.*s"
msgstr "на %.*s"

As you can see, the leading space is lost in translation.

By the way, po/ru.po as well as other po/*.po files are maintained elsewhere.

**Comment by mikhailnov** [comment]
Thank you. This issue seems to exist in other languages, but there is a question about languages where text goes from right to left, e.g. po/ta.po:

```
msgid " from %.*s"
msgstr "%.*s இலிருந்து"

msgid " on %.*s"
msgstr " %.*s இல்"
```
In one case there is a leading space, in another there is not, I can't understand why the space is in the beginning of the line, not after "இல்" in the end of it.

> are maintained elsewhere

I will make a pull request here, maybe it can be accepted without spending time on external services. I will have to leave fixing translations of right-to-left written languages to someone else as I don't understand how it must be correctly.

**Comment by ldv-alt** [comment]
Fixed in commit 41ac49ef2725f76b897da75a492a55dfef2cc920.
